master/internal/resourcemanagers/agent/agent.go

```diff
@@ -358,6 +359,7 @@ func (a *agent) containerStateChanged(ctx *actor.Context, sc aproto.ContainerSta
 		rsc.ContainerStarted = &sproto.TaskContainerStarted{
 			Addresses: sc.ContainerStarted.Addresses(),
 		}
+		prom.AssociateContainerRuntimeID(sc.Container.ID.String(), sc.ContainerStarted.ContainerInfo.ID)
```

Comment:
i think this would be better in `allocation.go` because if it's here this association won't work for k8s. i think maybe `sproto.TaskContainerStarted` should have a field like `NativeReservationID` that would be a container id for agents and a pod id for k8s (and so on), then the alloc can do this for each reservation (where reservation is just our abstract name or process/pod/container/etc one clump of resources)